packages/fx-core/src/plugins/resource/frontend/plugin.ts

```diff
@@ -169,23 +170,33 @@ export class FrontendPluginImpl {
     let runtimeEnv: RuntimeEnvironment | undefined;
     let aadEnv: AADEnvironment | undefined;
 
-    const functionPlugin = ctx.configOfOtherPlugins.get(DependentPluginInfo.FunctionPluginName);
-    if (functionPlugin) {
-      functionEnv = {
-        defaultName: functionPlugin.get(DependentPluginInfo.FunctionDefaultName) as string,
-        endpoint: functionPlugin.get(DependentPluginInfo.FunctionEndpoint) as string,
-      };
-    }
-
     if (isArmSupportEnabled()) {
-      const endpoint = getArmOutput(ctx, ArmOutput.SimpleAuthEndpoint) as string;
-      if (endpoint) {
+      const functionPlugin = ctx.configOfOtherPlugins.get(DependentPluginInfo.FunctionPluginName);
+      if (functionPlugin) {
+        const functionEndpoint = getArmOutput(ctx, FunctionArmOutput.Endpoint) as string;
+        functionEnv = {
+          defaultName: functionPlugin.get(DependentPluginInfo.FunctionDefaultName) as string,
+          endpoint: `https://${functionEndpoint}`,
```

Comment:
Looks we only need to calculate endpoint inside the if-else clause and then construct the functionEnv outside so we can reduce duplicate code.

Reply:
I think we can avoid sharing logic code between arm support and no arm support because they will not coexist if we switch to arm support. The current style is more readable. We can improve the whole code easily when we switch it.